sp_api/base/client.py

```diff
@@ -64,7 +65,9 @@ def auth(self) -> AccessTokenResponse:
 
     @property
     def grantless_auth(self) -> AccessTokenResponse:
-        return self._auth.get_grantless_auth()
+        if not self.grantless_scope:
+            raise Exception("Grantless operations require scope")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.

Reply:
Currently in the library there's `SellingApiException` with several classes deriving from it, `AuthorizationError` and `MissingCredentials` - both deriving from Exception directly.

Ideally we could introduce an exception class all other exceptions raised by the library could derive from. Such solution would make it easier to handle and log "our" exceptions - within the library and for 3rd party code using it.

Suppose it would look like this:
```
Exception (standard class)
|
SPApiException
|  \
|   \ 
|    \ 
|     MissingCredentials, AuthorizationError, BadUsageError (...)
SellingApiException
|
SellingApiBadRequestException, SellingApiForbiddenException, SellingApiNotFoundException(....)
```

`Client.grantless_auth` would then raise `BadUsageError` exception.

What do you think?